plugins/transport-nio/src/main/java/org/elasticsearch/http/nio/NioHttpServerTransport.java

```diff
@@ -83,30 +79,25 @@
 public class NioHttpServerTransport extends AbstractHttpServerTransport {
     private static final Logger logger = LogManager.getLogger(NioHttpServerTransport.class);
 
-    public static final Setting<Integer> NIO_HTTP_ACCEPTOR_COUNT =
-        intSetting("http.nio.acceptor_count", 1, 1, Setting.Property.NodeScope);
-    public static final Setting<Integer> NIO_HTTP_WORKER_COUNT =
-        new Setting<>("http.nio.worker_count",
-            (s) -> Integer.toString(EsExecutors.numberOfProcessors(s) * 2),
-            (s) -> Setting.parseInt(s, 1, "http.nio.worker_count"), Setting.Property.NodeScope);
-
     protected final PageCacheRecycler pageCacheRecycler;
     protected final NioCorsConfig corsConfig;
+    private final NioGroupFactory nioGroupFactory;
 
     protected final boolean tcpNoDelay;
     protected final boolean tcpKeepAlive;
     protected final boolean reuseAddress;
     protected final int tcpSendBufferSize;
     protected final int tcpReceiveBufferSize;
 
-    private NioGroup nioGroup;
+    private volatile SharedNioGroup nioGroup;
```

Comment:
why is it now volatile, was this missing?

Reply:
It probably should have been volatile since we access it from different threads. I assume that it incidentally worked in the past because there are probably other memory barriers between the write and usages.